### PR TITLE
Remove PY_CONDITION

### PR DESCRIPTION
**Pull request by isuruf**
Circle-CI and Travis-CI configs already does this. So, why not appveyor config?
cc @conda-forge/staged-recipes

### CODE REVIEW

**Comment by scopatz** [comment]
Aren't certain versions of Python tied to certain version of VS on Windows? I am not an expert here...

**Comment by isuruf** [comment]
But, conda-build-all and conda-build should do the correct thing when creating the envs. cc @msarahan

**Comment by msarahan** [comment]
I can't speak for what conda-build-all is doing with this.  It does seem like removing it would change what conda-build-all is doing here.  Those conditions are how we're getting one build per job.

Conda-build 3 won't be affected by this, but we might want to keep it around for getting one build per job like we do here.  The conda-smithy work I've done takes effect when a feedstock is rerendered/regenerated - I don't think it will affect things this early in the process.

**Comment by isuruf** [comment]
In travis-ci and linux we have only one job. I don't see what difference it makes. We might have needed this when we were having lots of numpy builds and one job couldn't handle everything due to timeouts. @jakirkham, @pelson what were the original reasons to separate the jobs by python version?

File: .appveyor.yml [hunk 1]
```diff
@@ -7,32 +7,11 @@ environment:
...
-      CONDA_INSTALL_LOCN: "C:\\Miniconda35"
-    - TARGET_ARCH: "x86"
-      CONDA_PY: "36"
-      PY_CONDITION: "python >=3.6"
       CONDA_INSTALL_LOCN: "C:\\Miniconda36"
+
     - TARGET_ARCH: "x64"
-      CONDA_PY: "27"
-      PY_CONDITION: "python >=2.7,<3"
       CONDA_INSTALL_LOCN: "C:\\Miniconda-x64"
```
Comment (xoviat) [inline .appveyor.yml]: This should be Miniconda3-x64.

**Comment by pelson** [comment]
> what were the original reasons to separate the jobs by python version?

As I recall, the main motivations were to control which python versions we build against. With the current PR, we would start building py26 binaries as far as I can see. In addition to that, conda-build-all does not handle visual studio as a build matrix dimension we only get one build of a particular compiled library, unless we explicitly mark it as depending upon Python (there is no mention of how we do the VC version matrix in the current docs, but for the first example I went to this wouldn't be a problem https://github.com/conda-forge/udunits2-feedstock/blob/master/recipe/meta.yaml as it does claim to depend on Python).


In summary, the reasons it was done like this were:
 * to explicitly control the python versions that we build against
 * to parallelise the builds for *mostly* separable sets of binaries
 * to ensure that non-python binaries get built against the necessary VC versions


Can anybody confirm whether these are still valid, and if not, I'm 👍  on merging.

**Comment by xoviat** [comment]
IMO this should be merged.

**Comment by isuruf** [comment]
> to explicitly control the python versions that we build against

There was an error and I fixed it in my PR. This is not a problem.

> to parallelise the builds for mostly separable sets of binaries

We run only 2 builds in parallel in appveyor, so having 6 or 2 doesn't matter

> to ensure that non-python binaries get built against the necessary VC versions

conda root env python version doesn't matter for conda-build. Am I correct @msarahan?

One of the main reasons for this PR is that we now have noarch and R recipes coming in, building those for 3 python versions is a waste.

**Comment by pelson** [comment]
> conda root env python version doesn't matter for conda-build. Am I correct @msarahan?

It isn't just that though. We actually need multiple builds of non-python binaries, one for each VC version. Do we have a policy that states that all compiled binaries must have python as a build dependency? (I can't remember).  If not, there is no magic anywhere that ensures we get a matrix for the VC versions.

**Comment by isuruf** [comment]
> We actually need multiple builds of non-python binaries, one for each VC version. Do we have a policy that states that all compiled binaries must have python as a build dependency? (I can't remember). If not, there is no magic anywhere that ensures we get a matrix for the VC versions.

We do have a recommendation at https://conda-forge.org/docs/buildwin.html#vc-features
We can't get a matrix for VC versions at staged-recipes without adding python as a build matrix yet. (We can with conda-build>=3)

**Comment by pelson** [comment]
OK, it's coming back to me now. Thanks :)

FWIW https://github.com/conda-tools/conda-build-all/pull/36 was an attempt to get conda-build-all to do a VC matrix without needing the python dependency hack, but I didn't get much chance to pursue it.

From what you've said, I see no reason not to merge this. I'm 👍  and would be happy if there is somebody else who is also 👍  and willing to merge.

**Comment by ocefpaf** [comment]
Done :smile:

**Comment by jakirkham** [comment]
Thanks @isuruf. Should be a nice improvement for the AppVeyor queue.

File: .appveyor.yml (hunk 1 above)
Comment (jakirkham) [reply .appveyor.yml]: Would you like to submit a follow-up PR to correct this?
